### ISSUE TITLE
Allow for connecting to a repl started with `clj`

### ISSUE THREAD

**Issue by slipset** [BetterThanTomorrow/calva#90]
https://github.com/clojure-emacs/cider-nrepl has instructions for how to fire up an cider-nrepl with `deps.edn`

When I try to make Calva connect to such an nrepl-session I get the following:
```
lein repl :connect localhost:50732
✔ ~/Documents/github.com/nrepl-clj
21:22 $ lein repl :connect localhost:50732
Connecting to nREPL at localhost:50732
CompilerException java.lang.ClassNotFoundException: clojure.tools.nrepl, compiling:(null:1:82)
#namespace[user]
Error loading namespace; falling back to user
nil
user=>
```

in the nrepl terminal.

deps.edn:

```clojure
{
 :aliases
 {:cider-clj {:extra-deps {org.clojure/clojure {:mvn/version "1.9.0"}
                           cider/cider-nrepl {:mvn/version "0.18.0"}
                           org.clojure/tools.nrepl {:mvn/version "0.2.13"}}
              :main-opts ["-m" "nrepl.cmdline" "--middleware" "[cider.nrepl/cider-middleware]"]}}}
```
Run as `clj -A:cider-clj`

**Comment by PEZ** [comment]
I am pretty sure I got Calva to connect, both the editor and the terminal REPLs, to this [deps-edn-figwheel project](https://github.com/PEZ/deps-edn-figwheel). Does that work for you?

**Comment by slipset** [comment]
D'oh! I'm not even reading the error message, even though it's right there!
`calva` uses `lein repl :connect` to connect to the nrepl server started with `clj`
It needs to use `clj` to connect to nrepl.

So, I made a `deps.edn` as such:
```
{
 :aliases
 {:nrepl-server {:extra-deps {org.clojure/tools.nrepl {:mvn/version "0.2.13"}
                              cider/cider-nrepl {:mvn/version "0.19.0-SNAPSHOT"}
                              refactor-nrepl {:mvn/version "2.3.1"}}
                 :main-opts ["-m" "nrepl.cmdline" "--middleware" "[cider.nrepl/cider-middleware]"]}
  :nrepl-client {:extra-deps {org.clojure/tools.nrepl {:mvn/version "0.2.13"}
                               cider/cider-nrepl {:mvn/version "0.19.0-SNAPSHOT"}
                               refactor-nrepl {:mvn/version "2.3.1"}}
                  :main-opts ["-m" "nrepl.cmdline" "--connect" "--host" "localhost" "--port"]}}}
```
Where `clj -A:nrepl-server` starts up a nrepl-server, and `clj -A:nrepl-client 4711` will connect to a nrepl server running on port 4711 on localhost. 

Given this, I'd suspect that `calva` could choose connection strategy depending on what files are present in the directory where `calva` is started, much like `cider` does.

**Comment by PEZ** [comment]
Cool! Can you make it so that everything goes on the command line w/o a need for `deps.edn` file for connecting? I tried and failed, but that was hopefully because I am not very familiar with `clj`.

**Comment by slipset** [comment]
```clj
clj -Sdeps '{:deps {org.clojure/tools.nrepl {:mvn/version "0.2.13"} 
                     cider/cider-nrepl {:mvn/version "0.19.0-SNAPSHOT"}
                     refactor-nrepl {:mvn/version "2.3.1"}}}' -m nrepl.cmdline --connect --host localhost --port 49832
```
Connects to a nrepl-server that's started with `clj -A:nrepl-server` from the `deps.edn` above.

I'm not sure if `refactor-nrepl` is needed on the client part of this.

**Comment by adairdavid** [comment]
I'm unable it seems to get this to work while providing in-line documentation and peek-definition support. Are you able to get that functionality working with this setup @slipset?

**Comment by alzadude** [comment]
FWIW, I am using `reply` instead of the built-in nrepl client, since the docs for nrepl recommend to use reply instead: https://nrepl.org/nrepl/usage/clients.html (refer to "Using the built-in client")

The `deps.edn` (minimal working example):
```
{:paths ["src"]
 :aliases {:nrepl-server {:extra-deps {cider/cider-nrepl {:mvn/version "0.18.0"}}
                          :main-opts ["-m" "nrepl.cmdline"]}}}
```
The command used to start the nrepl server:
```
clj -A:nrepl-server
```
The `calva.connectREPLCommand`:
```
clj -Sdeps '{:deps {reply {:mvn/version "0.4.1"}}}' -m reply.main --color --attach
```

**Comment by orestis** [comment]
@alzadude Can you actually use calva with this command? It works from the terminal, but from within Calva, I get:

```
clj -Sdeps '{:deps {reply {:mvn/version "0.4.1"}}}' -m reply.main --color --attach localhost:60991
Execution error at user/eval7044 (REPL:1).
No namespace: reply.eval-modes.nrepl found
REPL-y 0.4.1, nREPL 0.5.3
Clojure 1.10.0
Java HotSpot(TM) 64-Bit Server VM 1.8.0_152-b16
        Exit: Control+D or (exit) or (quit)
    Commands: (user/help)
        Docs: (doc function-name-here)
              (find-doc "part-of-name-here")
Find by Name: (find-name "part-of-name-here")
      Source: (source function-name-here)
     Javadoc: (javadoc java-object-or-class-here)
    Examples from clojuredocs.org: [clojuredocs or cdoc]
              (user/clojuredocs name-here)
              (user/clojuredocs "ns-here" "name-here")
WARNING: cat already refers to: #'clojure.core/cat in namespace: net.cgrand.parsley.fold, being replaced by: #'net.cgrand.parsley.fold/cat
user=>
``` 

Then when trying to evaluate a simple `(+ 1 2)` I get:

`Syntax error compiling at (REPL:1:1). Unable to resolve symbol: + in this context.`

**Comment by alzadude** [comment]
> Can you actually use calva with this command?

@orestis yes, it's working for me with the `deps.edn` in my comment, I can evaluate `(+ 1 2)` fine.

It's also working for me with the `deps.edn` in your comment (with the newer dependency versions).

So I can't understand why it's not working for you?

Actually I noticed my `deps.edn` didn't specify the cider middleware, I'm actually confused as to why it works without it, but for me it does..

For completeness I believe this is a more correct `deps.edn`, but for me it works regardless. I didn't specify `nrepl/nrepl` explicitly since `cider/cider-nrepl` depends on it, but it's probably better to make it explicit:
```
{:paths ["src"]
 :aliases {:nrepl-server {:extra-deps {cider/cider-nrepl {:mvn/version "0.18.0"}}
                          :main-opts ["-m" "nrepl.cmdline"
                                      "--middleware" "[cider.nrepl/cider-middleware]"]}}}
```

**Comment by cfehse** [comment]
This should be fixed with version 2.0.49.
